packages/launcher/src/renderer/launcher/developer/CreateAppModal.js

```diff
@@ -0,0 +1,463 @@
+
+  onFileInputChange = () => {
+    const files = [...this.fileInput.current.files]
+    if (files.length) {
+      this.setState(({ appData }) => ({
+        appData: {
+          ...appData,
+          contentsPath: files[0].path,
+        },
+      }))
+    }
+  }
+
+  onSetAppData = () => {
+    const { appData } = this.state
+    if (!appData.name || !appData.version || !appData.contentsPath) {
+      this.setState({
+        errorMsg: 'Please complete all fields.',
+      })
+    } else if (!isValidSemver(appData.version)) {
+      this.setState({
+        errorMsg: 'Please provide a valid verison number, e.g. 1.0.0',
+      })
+    } else {
+      this.setState({
+        errorMsg: undefined,
+        screen: 'identity',
+      })
+    }
+  }
+
+  onSetPermissions = (
+    permissionsRequirements: StrictPermissionsRequirements,
+  ) => {
+    this.setState({
+      permissionsRequirements,
+      screen: 'summary',
+    })
+  }
+
+  // RENDER
+
+  renderInfoForm() {
+    return (
+      <View>
+        <Text style={globalStyles.header}>App Info</Text>
+        <Text style={styles.description}>Set your app details.</Text>
+        <TextInput
+          placeholder="App name"
+          style={styles.input}
+          onChangeText={this.onChangeName}
+          value={this.state.appData.name}
+        />
+        <TextInput
+          placeholder="Version"
+          style={styles.input}
+          onChangeText={this.onChangeVersion}
+          value={this.state.appData.version}
+        />
+        <TouchableOpacity
+          onPress={this.onPressSelectContentsFolder}
+          style={styles.contentsPathButton}>
+          <Text style={styles.contentsPathLabel}>
+            {this.state.appData.contentsPath || 'Select Contents Path'}
+          </Text>
+        </TouchableOpacity>
+        <input
+          multiple
+          id="installer-file-selector"
+          onChange={this.onFileInputChange}
+          ref={this.fileInput}
+          type="file"
+          hidden
+          value={this.state.inputValue}
+        />
+        <Button title="CONTINUE" onPress={this.onSetAppData} />
+      </View>
+    )
+  }
+
+  renderSetIdentity() {
+    return (
+      <View>
+        <Text style={globalStyles.header}>Developer Identity</Text>
+        <IdentitySelectorView
+          enableCreate
+          type="developer"
+          onSelectId={this.onSelectId}
+          identities={this.state.devIdentities}
+          onCreatedId={this.onCreatedId}
+        />
+      </View>
+    )
+  }
+
+  renderPermissions() {
+    return (
+      <View>
+        <Text style={globalStyles.header}>App Permissions</Text>
+        <PermissionsRequirementsView onSetPermissions={this.onSetPermissions} />
+      </View>
+    )
+  }
+
+  renderSummary() {
+    const { appData, permissionsRequirements } = this.state
+
+    let permissionsContainer
+
+    const requirements = permissionsRequirements
+
+    if (requirements) {
+      const renderPermissions = (
```

Comment:
This render function should be a method in the class rather than being created every time `renderSummary()` is called.